### ISSUE TITLE
The Keyman script does not work on Mac in Chrome and Opera browsers

### ISSUE THREAD

**Issue by Laisvunas** [keymanapp/keyman#4496]
The Keyman script at https://keymanweb.com/ does not work on Mac in Chrome and Opera browsers, but works correctly on  Mac in Safari and Firefox browsers.

In Chrome and Opera browsers on Mac these errors is being outputted at https://keymanweb.com/ page:

Uncaught TypeError: Cannot read property 'length' of null
    at g.detect (kmwdevice.ts:92)
    at e.initDevices (kmwutils.ts:72)
    at new e (kmwutils.ts:51)
    at new k (kmwbase.ts:116)
    at kmwbase.ts:639
    at kmwbase.ts:631
kmwuitoolbar.ts:9 Uncaught TypeError: Cannot read property 'ui' of undefined
    at kmwuitoolbar.ts:9
    at kmwuitoolbar.ts:1190
kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11 Uncaught ReferenceError: keyman is not defined
    at getKeymanWeb (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11)
    at kmwlive.663f7399d98a234f17ddc68d22c8b805.js:386
(index):96 Uncaught ReferenceError: keyman is not defined
    at loadKeyboardFromHash ((index):96)
    at (index):127
helpers.ts:111 Uncaught ReferenceError: keyman is not defined
    at getKeymanWeb (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11)
    at HTMLDocument.<anonymous> (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:184)
    at j (jquery1-11-1.min.4dc834d16a0d219d5c2b8a5b814569e4.js:2)
    at Object.fireWith [as resolveWith] (jquery1-11-1.min.4dc834d16a0d219d5c2b8a5b814569e4.js:2)
    at Function.ready (jquery1-11-1.min.4dc834d16a0d219d5c2b8a5b814569e4.js:2)
    at HTMLDocument.J (jquery1-11-1.min.4dc834d16a0d219d5c2b8a5b814569e4.js:2)
    at HTMLDocument.sentryWrapped (helpers.ts:87)
2kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11 Uncaught ReferenceError: keyman is not defined
    at getKeymanWeb (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11)
    at updateSize (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:127)
kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11 Uncaught ReferenceError: keyman is not defined
    at getKeymanWeb (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11)
    at updateSize (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:127)

**Comment by mcdurdin** [comment]
@jahorton can you investigate please?

**Comment by jahorton** [comment]
I don't see this issue on my Mac (10.15.7) for either browser.

<img width="1259" alt="Screen Shot 2021-02-22 at 8 22 30 AM" src="https://user-images.githubusercontent.com/25213402/108646040-3fb49a00-74e7-11eb-948e-af834eb11b37.png">
<img width="1087" alt="Screen Shot 2021-02-22 at 8 23 17 AM" src="https://user-images.githubusercontent.com/25213402/108646047-417e5d80-74e7-11eb-8670-7694ddb3d5e2.png">

Looking at the code block for the error report, though... I have a suspicion that this may be related to macOS 11.x:

https://github.com/keymanapp/keyman/blob/72a9ac33c070ef0e4b46201e498f3aaf7c481dc5/web/source/kmwdevice.ts#L80-L91

The day has finally come where there's a version 11, after all.

**Comment by jahorton** [comment]
Yep, that seems to be the cause.  Used the debugger to replace my agent string with `Mozilla/5.0 (Macintosh; Intel Mac OS X 11_2_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.182 Safari/537.36`; that reproduced the error.

**Comment by mcdurdin** [comment]
We've merged a fix that should correct this problem. @Laisvunas can you re-test on keymanweb.com? (you may need to clear cache in order to get the latest version)

**Comment by Laisvunas** [comment]
I retested keymanweb.com and confirm that the issue has been fixed.

**Comment by mcdurdin** [comment]
Thank you! Closing this issue
